gdnative-core/src/variant.rs

```diff
@@ -884,4 +929,177 @@ impl FromVariant for Variant {
     fn from_variant(variant: &Variant) -> Option<Self> {
         Some(variant.clone())
     }
-}
\ No newline at end of file
+}
+
+impl<T> ToVariant for std::marker::PhantomData<T> {
+    fn to_variant(&self) -> Variant {
+        Variant::new()
+    }
+}
+
+impl<T> FromVariant for std::marker::PhantomData<T> {
+    fn from_variant(_: &Variant) -> Option<Self> {
+        Some(std::marker::PhantomData)
```

Comment:
Shouldn't this perform a check and only accept if the `Variant` is null?

Reply:
I think that could work as well.